### PR TITLE
V1.4.1 merge services

### PR DESCRIPTION
**Pull request by supritat**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### CODE REVIEW

File: jazz_services/components/validation/validate_update_payload.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        validateEmailFieldValue: function(onComplete) {
+            logger.info("Inside validateEmailFieldValue: ");
+            validateUtils.validateEmail(service_data, onComplete);
+        },
+
+        validateStatusStateChange: function(onComplete) {
+            logger.info("Inside validateStatusStateChange: ");
+            validateUtils.validateStatusStateChange(service_data, service_data_from_db, onComplete);
+        }
+        // ,
```
Comment (deepusundar) [inline jazz_services/components/validation/validate_update_payload.js]: Clean up the commented code

Comment (supritat) [reply jazz_services/components/validation/validate_update_payload.js]: Done

File: jazz_services/components/utils.js [hunk 2]
```diff
@@ -181,13 +181,59 @@ var paginateUtil = function(data, limit, offset) {
...
+        data = data.sort(function(a, b) {
+            var x = a[sort_key];
+            var y = b[sort_key];
+            if (sort_direction === "asc") return x < y ? -1 : x > y ? 1 : 0;
+            else {
+                return x < y ? 1 : x > y ? -1 : 0;
+            }
+        });
+    } else {
+        console.log('sort key undefined')
```
Comment (suryajak) [inline jazz_services/components/utils.js]: remove console log

File: jazz_services/components/utils.js [hunk 3]
```diff
@@ -181,13 +181,59 @@ var paginateUtil = function(data, limit, offset) {
     return data;
 };
 
+var sortUtil = function(data, sort_key, sort_direction) {
+    if (sort_key !== undefined && sort_key !== "timestamp") {
+        console.log('sort key defined')
```
Comment (suryajak) [inline jazz_services/components/utils.js]: remove console log

File: jazz_services/components/utils.js [hunk 4]
```diff
@@ -181,13 +181,59 @@ var paginateUtil = function(data, limit, offset) {
...
+            var y = new Date(val2).getTime();
+            if (sort_direction === "asc") return x < y ? -1 : x > y ? 1 : 0;
+            else return x < y ? 1 : x > y ? -1 : 0;
+        });
+    }
+    return data;
+};
+
+var filterUtil = function(data, filter_value) {
+    console.log('filter util called....')
```
Comment (suryajak) [inline jazz_services/components/utils.js]: remove console log

File: jazz_services/components/validation/validate_update_payload.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    });
+
+    var service_data_from_db = {};
+
+    async.series({
+        validateServiceExists: function(onComplete) {
+            // logger.info('crud.get ')
+            crud.get(service_id, function onServiceGet(error, data) {
+                if (error) {
+                    logger.info('crud.get error')
```
Comment (suryajak) [inline jazz_services/components/validation/validate_update_payload.js]: logger.error. Also log the error itself

File: jazz_services/index.js [hunk 6]
```diff
@@ -73,45 +95,49 @@ module.exports.handler = (event, context, cb) => {
...
                 // Handle error
                 if (error) {
                     logger.error('Error occured. ' + JSON.stringify(error, null, 2));
-                    return cb(JSON.stringify(errorHandler.throwInternalServerError('unexpected error occured')));
+                    handleResponse(error, data.getServiceByServiceId, event.query)
                 }
-
-                return cb(null, responseObj(result.fetchServices, event.query));
+                if (data && data.services.length === 0 && event.query) {
+                    logger.error("Cannot find service with query: " + JSON.stringify(event.query));
```
Comment (suryajak) [inline jazz_services/index.js]: Why is this error? This is at best warn but mostly info

File: jazz_services/index.js [hunk 7]
```diff
@@ -124,98 +150,35 @@ module.exports.handler = (event, context, cb) => {
...
                     }
                 }
             }, function(error, data) {
                 // Handle error
-                logger.info('error');
-                logger.info(error);
-                logger.info(data);
                 if (error) {
-                    logger.error('error occured while updating service: ' + service_id);
+                    logger.error('validateServiceExists ' + service_id);
```
Comment (suryajak) [inline jazz_services/index.js]: combine the two logger.error into 1

File: jazz_services/components/utils.js (hunk 2 above)
Comment (supritat) [reply jazz_services/components/utils.js]: removed console log

File: jazz_services/components/utils.js (hunk 3 above)
Comment (supritat) [reply jazz_services/components/utils.js]: removed console log

File: jazz_services/components/utils.js (hunk 4 above)
Comment (supritat) [reply jazz_services/components/utils.js]: removed console log

File: jazz_services/components/validation/validate_update_payload.js (hunk 5 above)
Comment (supritat) [reply jazz_services/components/validation/validate_update_payload.js]: done

File: jazz_services/index.js (hunk 6 above)
Comment (supritat) [reply jazz_services/index.js]: changes are done as mentioned

File: jazz_services/index.js (hunk 7 above)
Comment (supritat) [reply jazz_services/index.js]: Done

**Pull request merged**